### PR TITLE
reouting/refactor Location does not need to be generic

### PR DESCRIPTION
**Pull request by dirvine**
Location did not need to be generic.

### CODE REVIEW

File: src/messages/mod.rs [hunk 1]
```diff
@@ -499,66 +499,10 @@ impl RoutingMessage {
...
-/// `ConnectionInfo`.
-///
-///
-/// ### Resource Proof Evaluation to approve
-/// When nodes Z of section Y receive `CandidateInfo` from A, they reply with a `ResourceProof`
-/// request. Node A needs to answer these requests (resolving a hashing challenge) with
-/// `ResourceProofResponse`. Members of Y will send out `CandidateApproval` messages to vote for the
-/// approval in their section. Once the vote succeeds, the members of Y send `NodeApproval` to A and
-/// add it into their routing table. When A receives the `NodeApproval` message, it adds the members
-/// of Y to its routing table.
```
Comment (jeanphilippeD) [inline src/messages/mod.rs]: did you mean to remove that comment as part of that commit

File: src/messages/mod.rs [hunk 2]
```diff
@@ -499,66 +499,10 @@ impl RoutingMessage {
...
 #[derive(Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
 // FIXME - See https://maidsafe.atlassian.net/browse/MAID-2026 for info on removing this exclusion.
 #[allow(clippy::large_enum_variant)]
+/// Content
```
Comment (jeanphilippeD) [inline src/messages/mod.rs]: did you mean to add that comment as part of that commit?

Comment (dirvine) [reply src/messages/mod.rs]: Ah yes, the atlassian link is prob not for everyone, but also clippy itself explains what this is. No need fo rus to repeat the possible solutions. I felt it did not add much to direct folk to atlassian.

File: src/messages/mod.rs (hunk 1 above)
Comment (dirvine) [reply src/messages/mod.rs]: Yes, it was redundant as we don't do the resource proof anymore.  The intention is to go through all docs soon and update them.

**Pull request merged**